packages/account/src/providers/provider.ts

```diff
@@ -775,6 +774,46 @@ export default class Provider {
     };
   }
 
+  /**
+   * Estimates the transaction gas and fee based on the provided transaction request.
+   * @param transactionRequest - The transaction request object.
+   * @param optimizeGas - Optional. Specifies whether to optimize the gas. Default is false.
+   * @returns An object containing the estimated minimum gas, minimum fee, maximum gas, and maximum fee.
+   */
+  estimateTxGasAndFee(params: { transactionRequest: TransactionRequest; optimizeGas?: boolean }) {
```

Comment:
Should we have a test for this function?

Reply:
Hasn't this already been addressed by testing the `getTransactionCost` method? 🤔

Reply:
@arboleya indeed. This method is being executed by almost all `e2e` tests. 

We also already have unit tests for the methods that it is using: https://github.com/FuelLabs/fuels-ts/blob/master/packages/account/src/providers/utils/gas.ts